### DISCUSSION TITLE
Setting Validation in Traits

### DISCUSSION THREAD

**Discussion by goldmerc** [livewire/livewire#4035] [3. Feature ideas]
If you want to add to the $queryString or the $listeners array in a trait, it's easy because these properties are initialised as empty arrays which you can just add to.

But for rules, messages & validationAttributes, this is not the case. The properties don't exist unless they've been initialised in your component.  The only way I can see to get around this is to override the get method for these properties. For instance...
```
    protected function getRules()
    {
        return array_merge([
            'minPrice' => 'bail|nullable|numeric|integer|min:0',
            'maxPrice' => "bail|nullable|numeric|integer|min:1",
        ], parent::getRules());
    }
```
But this example will only work, if you only have one trait which needs to add rules.

Is there a reason why these properties are not simply declared as empty arrays in the Livewire ValidatesInput trait?

**Comment by joshhanley** [comment]
@goldmerc yeah that would be useful. I don't think there is any specific reason it's setup that way currently. PR's are welcome, as long as it's not a breaking change 🙂
